sdk/core/Azure.Core/src/Pipeline/RequestFailedDetailsParser.cs

```diff
@@ -14,6 +14,9 @@ public abstract class RequestFailedDetailsParser
         /// <summary>
         /// Parses the error details from the provided <see cref="Response"/>.
         /// </summary>
+        /// <remarks>
+        /// In various scenarios, parsers can get called for successful responses. Implementations should not populate <paramref name="error"/> or <paramref name="data"/> with sensitive information, as these values may be logged as part of the exception.
```

Comment:
Super nit:

```suggestion
        /// In various scenarios, parsers may be called for successful responses. Implementations should not populate <paramref name="error"/> or <paramref name="data"/> with sensitive information, as these values may be logged as part of the exception.
```

But otherwise, ❤️ it! :)

Reply:
thanks!